src/core/thread/mesh_forwarder.cpp

```diff
@@ -2016,6 +2016,11 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const Thre
     child->SetLastHeard(Timer::GetNow());
     child->ResetLinkFailures();
 
+    if (aMacSource.mLength == sizeof(aMacSource.mShortAddress))
+    {
+        mSourceMatchController.SetSrcMatchAsShort(*child, true);
```

Comment:
My point is if a Child fails to receive an MLE Child ID Response, it has no choice but to transmit an MLE Child ID Request again (or restart the attach process from the MLE Parent Request).